src/cluster.py

```diff
@@ -710,6 +731,57 @@ def primary_changed(self, old_primary: str) -> bool:
         primary = self.get_primary()
         return primary != old_primary
 
+    def remove_raft_data(self) -> None:
```

Comment:
We can most probably converge this with https://github.com/canonical/postgresql-operator/blob/main/src/relations/async_replication.py#L695 as a followup